### PR TITLE
swarm / rcmgr: synchronize the concurrent outbound dials with limits

### LINKED ISSUE #1668: circuitv2: flaky TestReservationFailures
https://github.com/libp2p/go-libp2p/runs/7748150365?check_suite_focus=true

```
  === RUN   TestReservationFailures
  === RUN   TestReservationFailures/unsupported_protocol
  2022/08/09 14:31:47 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/lucas-clemente/quic-go/wiki/UDP-Receive-Buffer-Size for details.
  === RUN   TestReservationFailures/wrong_message_type
  === RUN   TestReservationFailures/unknown_status
  === RUN   TestReservationFailures/invalid_time
      reservation_test.go:101: 
          	Error Trace:	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/circuitv2/client/reservation_test.go:101
          	Error:      	"error writing reservation message: Application error 0x0: peer:QmQkiUAs4viJ4L6VYx1f4QTcc8oWcCsk8qxDxU9efYK3Fs: cannot reserve inbound connection: resource limit exceeded" does not contain "received reservation with expiration date in the past"
          	Test:       	TestReservationFailures/invalid_time
  === RUN   TestReservationFailures/invalid_voucher
  --- FAIL: TestReservationFailures (8.07s)
      --- PASS: TestReservationFailures/unsupported_protocol (2.04s)
      --- PASS: TestReservationFailures/wrong_message_type (1.53s)
      --- PASS: TestReservationFailures/unknown_status (1.77s)
      --- FAIL: TestReservationFailures/invalid_time (1.72s)
      --- PASS: TestReservationFailures/invalid_voucher (1.01s)

### LINKED ISSUE #1730: circuitv2: flaky TestReservationFailures/unsupported_protocol
```
  === RUN   TestReservationFailures/unsupported_protocol
      reservation_test.go:100: 
          	Error Trace:	D:\a\go-libp2p\go-libp2p\p2p\protocol\circuitv2\client\reservation_test.go:100
          	Error:      	"Application error 0x0: peer:12D3KooWDuwgsbf9CcVe1euAXgjn8CxtHKjQdhuLGiJ7Na4NWinE: cannot reserve inbound connection: resource limit exceeded" does not contain "protocol not supported"
          	Test:       	TestReservationFailures/unsupported_protocol
  === RUN   TestReservationFailures/wrong_message_type
  === RUN   TestReservationFailures/unknown_status
  === RUN   TestReservationFailures/invalid_time
  === RUN   TestReservationFailures/invalid_voucher
  --- FAIL: TestReservationFailures (0.64s)
      --- FAIL: TestReservationFailures/unsupported_protocol (0.20s)
      --- PASS: TestReservationFailures/wrong_message_type (0.09s)
      --- PASS: TestReservationFailures/unknown_status (0.08s)
      --- PASS: TestReservationFailures/invalid_time (0.17s)
      --- PASS: TestReservationFailures/invalid_voucher (0.10s)

### LINKED ISSUE #1765: circuitv2: flaky TestReservationFailures
https://github.com/libp2p/go-libp2p/actions/runs/3092616835/jobs/5004089751

```
  === RUN   TestReservationFailures/unsupported_protocol
      reservation_test.go:100: 
          	Error Trace:	D:\a\go-libp2p\go-libp2p\p2p\protocol\circuitv2\client\reservation_test.go:100
          	Error:      	"Application error 0x0: peer:12D3KooWKfsfaScKKyfhMAsdiRPKqDiK7TYWf5Lq8EmLXKELS67X: cannot reserve inbound connection: resource limit exceeded" does not contain "protocol not supported"
          	Test:       	TestReservationFailures/unsupported_protocol

### LINKED ISSUE #1816: circuitv2: flaky TestReservationFailures
https://github.com/libp2p/go-libp2p/actions/runs/3208002096/jobs/5243462098

```
  === RUN   TestReservationFailures/invalid_time
      reservation_test.go:100: 
          	Error Trace:	/home/runner/work/go-libp2p/go-libp2p/p2p/protocol/circuitv2/client/reservation_test.go:100
          	Error:      	"error writing reservation message: Application error 0x0: peer:12D3KooWQMrNzVJdPSsgvtwFvZ4jPLYNjXZ1Xc4BAPWtA4MGAtMy: cannot reserve inbound connection: resource limit exceeded" does not contain "received reservation with expiration date in the past"
          	Test:       	TestReservationFailures/invalid_time
  === RUN   TestReservationFailures/invalid_voucher
  --- FAIL: TestReservationFailures (0.89s)
      --- PASS: TestReservationFailures/unsupported_protocol (0.20s)
      --- PASS: TestReservationFailures/wrong_message_type (0.14s)
      --- PASS: TestReservationFailures/unknown_status (0.17s)
      --- FAIL: TestReservationFailures/invalid_time (0.18s)
      --- PASS: TestReservationFailures/invalid_voucher (0.20s)

### PR DESCRIPTION
**Pull request by MarcoPolo**
We see flakiness in CI because a client will dial with concurrency 8 but the limit on the server is 4 so we get rcmgr failures. We notice this when we have a lot of addrs to dial.

 Fixes #1816
 Fixes #1765
 Fixes #1730
 Fixes #1668
 
 (will rebase to master after https://github.com/libp2p/go-libp2p/pull/1881 is merged)

### CODE REVIEW

**Comment by marten-seemann** [comment]
> (will rebase to master after https://github.com/libp2p/go-libp2p/pull/1881 is merged)

Why not target `master` from this PR? Otherwise this commit will be lost when we squash #1881.

**Comment by MarcoPolo** [comment]
> > (will rebase to master after #1881 is merged)
> 
> Why not target `master` from this PR? Otherwise this commit will be lost when we squash #1881.

To make it easier to review. I'll merge this to master so we'll keep this commit. I should have said target master after #1881

**Comment by MarcoPolo** [comment]
Oh I see what you meant. This PR didn't build on the quic-v1 branch so there was no point in targeting that branch. I just defaulted to stacking it since it came from that branch, but I should have just based this on master from the start.

**Comment by MarcoPolo** [comment]
@marten-seemann and I talked about this and agreed to bump the limit up to 8 for now. This will be handled better when we have smart dialing.

File: p2p/host/resource-manager/limit_defaults.go [hunk 1]
```diff
@@ -492,7 +492,7 @@ var DefaultLimits = ScalingLimitConfig{
 	},
 
 	PeerBaseLimit: BaseLimit{
-		ConnsInbound:    4,
+		ConnsInbound:    8,
```
Comment (marten-seemann) [inline p2p/host/resource-manager/limit_defaults.go:495]: Can you add a comment explaining why this is 8 for now, and that smart dialing will allow us to reduce this back to 4?

**Review by marten-seemann — APPROVED** [review]

**Pull request merged**